### PR TITLE
Compare filters clean up

### PR DESCRIPTION
**Pull request by chasulin**
Adds `useCompareSliderValueMap` hook to update the MUI Slider component's numerical value whenever the filter selection changes on homepage or location page. We were previously passing `setHomepageSliderValue` and `setSliderValue` down as props and doing this update in multiple places in the compare code. This unifies into a single hook that is called once from the top (in `CompareMain`)

### CODE REVIEW

File: src/common/utils/compare.ts [hunk 1]
```diff
@@ -365,3 +365,34 @@ export const getCompareSubheader = (region: Region): string => {
     return 'Compare counties';
   }
 };
+
+// Value maps for filters' slider components (0, 50, and 99 are numerical values required by MUI Slider).
```
Comment (chasulin) [inline src/common/utils/compare.ts:369]: These maps were previously in Filter.tsx and CompareMain.tsx. Moved them all here

File: src/components/Compare/CompareMain.tsx [hunk 2]
```diff
@@ -175,15 +163,16 @@ const CompareMain = (props: {
...
-  // Location page slider:
-  const defaultSliderValue = scopeValueMap[geoScope];
-  const [sliderValue, setSliderValue] = useState(defaultSliderValue);
-
-  // Homepage slider:
-  const defaultHomepageSliderValue = homepageScopeValueMap[homepageScope];
-  const [homepageSliderValue, setHomepageSliderValue] = useState(
-    defaultHomepageSliderValue,
-  );
+  // Since the route isn't changing when navigating between county pages within the same state, state variables weren't resetting. This forces a reset:
```
Comment (chasulin) [inline src/components/Compare/CompareMain.tsx:166]: This useEffect is already in develop, but **this current** PR's branch is based off of the branch where the effect was added, so the change is showing up here as well

File: src/common/hooks/useHomepageCompareSliderMap.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  homepageScope: HomepageLocationScope,
+) {
+  const defaultHomepageSliderValue = homepageScopeValueMap[homepageScope];
+  const [homepageSliderValue, setHomepageSliderValue] = useState(
+    defaultHomepageSliderValue,
+  );
+
+  useEffect(() => {
+    setHomepageSliderValue(homepageScopeValueMap[homepageScope]);
+  }, [homepageScope]);
```
Comment (pnavarrc) [inline src/common/hooks/useHomepageCompareSliderMap.ts:22]: I think `setHomepageSliderValue` needs to be a dependency as well

File: src/common/hooks/useLocationPageCompareSliderMap.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+export default function useLocationPageCompareSliderMap(
+  geoScope: GeoScopeFilter,
+) {
+  const defaultSliderValue = scopeValueMap[geoScope];
+  const [sliderValue, setSliderValue] = useState(defaultSliderValue);
+
+  useEffect(() => {
+    setSliderValue(scopeValueMap[geoScope]);
+  }, [geoScope]);
```
Comment (pnavarrc) [inline src/common/hooks/useLocationPageCompareSliderMap.ts:17]: `setSliderValue` should be a dependency here too

**Review by pnavarrc — APPROVED** [review]
Looks good!

File: src/common/hooks/useHomepageCompareSliderMap.ts (hunk 3 above)
Comment (chasulin) [reply src/common/hooks/useHomepageCompareSliderMap.ts:22]: hm the linter doesnt complain when its omitted. just double checked [in the docs](https://reactjs.org/docs/hooks-reference.html) - ("React guarantees that setState function identity is stable and won’t change on re-renders. This is why it’s safe to omit from the useEffect or useCallback dependency list.")

File: src/common/hooks/useLocationPageCompareSliderMap.ts (hunk 4 above)
Comment (chasulin) [reply src/common/hooks/useLocationPageCompareSliderMap.ts:17]: same as above^

File: src/common/hooks/useHomepageCompareSliderMap.ts (hunk 3 above)
Comment (pnavarrc) [reply src/common/hooks/useHomepageCompareSliderMap.ts:22]: nice!

**Pull request merged**